File: slack-reader/src/main/java/dev/number6/slackreader/SlackReaderRequestHandler.java
package dev.number6.slackreader;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import dev.number6.slackreader.dagger.DaggerSlackReaderComponent;

import java.util.Map;

public class SlackReaderRequestHandler implements RequestHandler<Map<String, Object>, String> {

    private final SlackReader handler = DaggerSlackReaderComponent.create().handler();

    @Override
    public String handleRequest(Map<String, Object> o, Context context) {
        handler.handle(o, context.getLogger());
        return "ok";
    }
}

File: slack-reader/src/main/kotlin/dev/number6/slackreader/SlackReaderRequestHandler.kt
package dev.number6.slackreader

import com.amazonaws.services.lambda.runtime.Context
import com.amazonaws.services.lambda.runtime.RequestHandler
import dev.number6.slackreader.dagger.DaggerSlackReaderComponent

class SlackReaderRequestHandler : RequestHandler<Map<String, Any>, String> {
    private val handler = DaggerSlackReaderComponent.create().handler()
    override fun handleRequest(o: Map<String, Any>, context: Context): String {
        handler.handle(o, context.logger)
        return "ok"
    }
}